### PR TITLE
executor: support turn on/off slow-log by variables

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Support turn on/off slow-log by variables.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: executor/adapter.go [hunk 1]
```diff
@@ -711,7 +711,8 @@ var QueryReplacer = strings.NewReplacer("\r", " ", "\n", " ", "\t", " ")
 
 func (a *ExecStmt) logAudit() {
 	sessVars := a.Ctx.GetSessionVars()
-	if sessVars.InRestrictedSQL {
+	enable := atomic.LoadUint32(&config.GetGlobalConfig().Log.EnableAuditLog) > 0
+	if enable && sessVars.InRestrictedSQL {
```
Comment (lamxTyler) [inline executor/adapter.go]: If `enable` is false, it will do audit log now?

**Comment by qw4990** [comment]
> The audit-log could be turned off by the plugin framework. And the slow-log could be turned off if the threshold is very large. I don't know the purpose of this PR.

Oh sorry for that, I don't know there is already a way to turn on/off audit logs(`admin enable/disable`).

As for slog logs, after discussing with our PM, he thinks it's better to have an explicit switch to turn on/off slow logs, instead of using the threshold variable, which would confuse users.

So how about removing the audit log switch and keep the slow log switch?

**Comment by jackysp** [comment]
emmm... For the slow logs, I think it is better to add a variable only, I think TiDB prefers to use fewer config items for the server which is also our PM requirements. But it is also wired to use a session variable to control the server scope behaviors, which is also our PM's concern. I'm a little confused about our PM's requirements.

**Comment by qw4990** [comment]
Session variables are easy to use since they can be modified by SQLs and this way allows users to turn on/off slow logs dynamically. These two features are required by our PM...

I agree with the point that `it is also wired to use a session variable to control the server scope behaviors`, but it's unfortunate that TiDB now has no server-scope-level variables. And many items already behave in this way... for example, `TiDBGeneralLog`, `TiDBSlowLogThreshold`, `TiDBRecordPlanInSlowLog`, `TiDBDDLSlowOprThreshold`, `TiDBQueryLogMaxLen`, `TiDBRetryLimit`...  🤦‍♂ 

Except for using session variables, would you have any other suggestions that can meet our PM's requirements?  @jackysp  PTAL...

**Comment by qw4990** [comment]
After discussing with @jackysp privately, we decide to remove the audit log switch and keep the slow log switch.
@lamxTyler PTAL

File: executor/adapter.go (hunk 1 above)
Comment (qw4990) [reply executor/adapter.go]: You are right, it's wrong.
And after discussing, this audit log switch has been removed now.

File: executor/adapter.go [hunk 2]
```diff
@@ -752,7 +752,8 @@ func (a *ExecStmt) LogSlowQuery(txnTS uint64, succ bool, hasMoreResults bool) {
 	cfg := config.GetGlobalConfig()
 	costTime := time.Since(sessVars.StartTime) + sessVars.DurationParse
 	threshold := time.Duration(atomic.LoadUint64(&cfg.Log.SlowThreshold)) * time.Millisecond
-	if costTime < threshold && level > zapcore.DebugLevel {
+	enable := atomic.LoadUint32(&cfg.Log.EnableSlowLog) > 0
+	if enable && costTime < threshold && level > zapcore.DebugLevel {
```
Comment (lamxTyler) [inline executor/adapter.go]: I am also confused by this, If enable is `false`, it will print slow logs.

Comment (qw4990) [reply executor/adapter.go]: Sorry, my mistake. It is fixed now.

File: sessionctx/variable/tidb_vars.go [hunk 3]
```diff
@@ -112,6 +112,9 @@ const (
 	// tidb_record_plan_in_slow_log is used to log the plan of the slow query.
 	TiDBRecordPlanInSlowLog = "tidb_record_plan_in_slow_log"
 
+	// tidb_enable_slow_log enables TiDB to log slow queries.
+	TiDBEnableSlowLog = "tidb_enable_slow_log"
```
Comment (jackysp) [inline sessionctx/variable/tidb_vars.go]: I still have some concerns about adding a variable. When we think that the variable is inappropriate, we cannot easily remove it, which will cause compatibility problems. It doesn't matter if you just add an item to the configuration file. It's better that we only add a configuration item. This also meets the requirements of PM, and there is no risk. :)

Comment (SunRunAway) [reply sessionctx/variable/tidb_vars.go]: Mysql has similar variable too.
```mysql
mysql> show variables like '%slow_query_log%';
+---------------------+-----------------------------------------------------------+
| Variable_name       | Value                                                     |
+---------------------+-----------------------------------------------------------+
| slow_query_log      | OFF                                                       |
| slow_query_log_file | /usr/local/var/mysql@8.0/fengliyuans-MacBook-Pro-slow.log |
+---------------------+-----------------------------------------------------------+
2 rows in set (0.00 sec)
```

Comment (jackysp) [reply sessionctx/variable/tidb_vars.go]: Should we use the same name as MySQL?

Comment (SunRunAway) [reply sessionctx/variable/tidb_vars.go]: I'd like to.

**Comment by lysu** [comment]
LGTM

**Pull request merged**